src/include/sof/trace/preproc.h

```diff
@@ -102,8 +102,8 @@
  * results in ADD(1,2) ADD(3,4) ADD(5,6) and so on
  * MAP##N must exist for arg_count == N to work
  */
-#define META_MAP(arg_count, m, ...)\
-	META_CONCAT(_META_MAP_, arg_count)(m, __VA_ARGS__)
+#define META_MAP(arg_count, m, ...) META_RECURSE(\
```

Comment:
poor compiler, my heart hurts when I see what `META_RECURSE()` does ;-) Maybe you could have some mercy over it and use just `META_RECURSE_2()` or at most `_4`? `META_RECURSE()` isn't used otherwise, do we really need 1024 levels of recursion for anything? `META_SEQ_FROM_0_TO()`, `META_CONCAT_SEQ_DELIM_()`, and `META_CONCAT_SEQ()`, which use `META_RECURSE()` don't seem to be used either.

Reply:
preprocess recurse stops when didn't find any change, so it's rather maximal recursion level, and preprocessor stops work much earlier as usual. I'm not sure right now, but 1024 means up to 10 level of indentation (2^10 = 1024).

Reply:
The preprocessing depth here should be limited (and all other places too). All too often we see one error recurse into 1024 errors which overflows the terminal and is difficult to debug. Can we delete ```META_RECURSE()```  and replace it with limited versions. I don't see any users here where we need > 8 levels ?

Reply:
Reduced default `META_RECURSE` depth to 8 and pulled last commit from @lyakh PR